### ISSUE TITLE
Jerky motion in light house of image gallary

### ISSUE THREAD

**Issue by NicestRudeGuy** [rtCamp/rtMedia#1699]
This is kind of irritating and might cause issues with the **User Experience** .

![Kapture 2020-10-22 at 12 33 31](https://user-images.githubusercontent.com/9863419/96860414-5c30bf00-1480-11eb-8d92-df3493c73c5d.gif)

**Possible solutions**  
1. Instead of re-rendering whole light house just update the media ( image/video/etc ) and the form values on the right side. 

2. Add fade in/out animation when media changes.

**Comment by nitun** [comment]
Good suggestion 👍 
Would you like to do it? if yes, please assign it to yourself. 

cc @pooja-muchandikar can help to test it out throughout the add-ons.

**Comment by NicestRudeGuy** [comment]
@nitun  before commiting to this issue i would need to check the working of a the rtMedia plugin .
i dont have experience with normal plugins so if i make changes i might break some stuff . 
if this was gutnberg related it would be easy to fix for me since im gaining experience in that day by day.

**Comment by nitun** [comment]
I totally understand you @NicestRudeGuy No worries.
@pooja-muchandikar please add this task in the to do for future releases. Maybe some other dev can work on it.

**Comment by pooja-muchandikar** [comment]
Sure

**Comment by NicestRudeGuy** [comment]
### Progress of today ( 2:25 pm onwards )

Read the documentation of rtMedia plugin from handbook
https://handbook.rtcamp.com/rtmedia/

Features from 
https://rtmedia.io/docs/features/

### Obersations

Read the code of lightbox , found that we are using magnific popup jquery . 
The problem is the jerky motion when  media (image/video) changes along with the right panel which house's the information regarding the media, although there is loader behind still the motion isn't smooth. 

After reading the magnific popup library found that it already have fade in and fade out animation which can solve this issue. 
but it the sidepanel can cause issue. 

So thinking of having custom css fade animation when media changes along with sidepanel. 

### Blockers 

Issue with the task runner ( grunt ) the code isnt compiling ,tried global version of sass and  tried several versions of node ( 14,12,10 ) tried removing **node_modules** folder and **package-lock.json** 
@SID177  also gave some links but its not working .

**Comment by NicestRudeGuy** [comment]
Got it working with **Node 12.16.1** .

**Comment by NicestRudeGuy** [comment]
Going with the fade animation solution.  
First will try to use the builtin animation of magnific popup if it doesnt work as expected then will have to write add custom animation. 
### ETA Today

**Comment by NicestRudeGuy** [comment]
### Final result

Tested on several browsers chrome, firefox , safari with mobile and tab devices as well. 

![animation](https://user-images.githubusercontent.com/9863419/99064932-a6fbad80-25cc-11eb-94cc-c26ff490cde5.gif)

**Comment by pooja-muchandikar** [comment]
Tested the change with `GH-1699` branch. And as an enhancement fading effect is visible on the lightbox. 

```
Test Environment
WordPress: v5.6
BuddyPress: v7.1.0
rtMedia: v4.6.5 GH-1699 branch
```

**Comment by pooja-muchandikar** [comment]
Tested with `pre-release` branch and the changes looks fine.

**Comment by pooja-muchandikar** [comment]
Tested with `develop` branch and the changes looks fine.

**Comment by pooja-muchandikar** [comment]
Released with `v4.6.6`
